explorer/src/entity/Admin.ts

```diff
@@ -17,3 +18,18 @@ export class Admin {
   @Column()
   updatedAt: Date
 }
+
+export function find(db: Connection, username: string): Promise<Admin> {
+  return db.getRepository(Admin).findOne({ username: username })
+}
+
+export async function isValidPassword(
+  password: string,
+  admin?: Admin,
+): Promise<boolean> {
+  if (!admin) {
+    return false
+  }
+
+  return await comparePassword(password, admin.hashedPassword)
```

Comment:
dont really need to await here, so then we dont need the `async` modifier either

Reply:
Isn't the `async` required to return a promise for L31?

Reply:
Nice catch, you’re right